File: lib/Infralution.Localization.Wpf/SampleApp_VB/SampleEnum.vb
#Region "File Header"
'
' COPYRIGHT:   Copyright 2009 
'              Infralution
'
#End Region
Imports System.ComponentModel
Imports Infralution.Localization.Wpf

''' <summary>
''' Sample enum illustrating used of a localized enum type converter
''' </summary>
<TypeConverter(GetType(SampleEnumConverter))> _
Public Enum SampleEnum
    VerySmall
    Small
    Medium
    Large
    VeryLarge
End Enum

''' <summary>
''' Define the type converter for the Sample Enum
''' </summary>
''' <remarks></remarks>
Class SampleEnumConverter
    Inherits ResourceEnumConverter

    ''' <summary>
    ''' Create a new instance of the converter using translations from the given resource manager
    ''' </summary>
    ''' <remarks></remarks>
    Public Sub New()
        MyBase.New(GetType(SampleEnum), My.Resources.ResourceManager)
    End Sub
End Class

File: lib/Infralution.Localization.Wpf/SampleApp_CS/SampleEnum.cs
#region File Header
//
// COPYRIGHT:   Copyright 2009 
//              Infralution
//
#endregion
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel;
using Infralution.Localization.Wpf;
namespace WpfApp
{

    /// <summary>
    /// Sample enum illustrating used of a localized enum type converter
    /// </summary>
    [TypeConverter(typeof(SampleEnumConverter))]
    public enum SampleEnum
    {
        VerySmall,
        Small,
        Medium,
        Large,
        VeryLarge
    }

    /// <summary>
    /// Define the type converter for the Sample Enum
    /// </summary>
    class SampleEnumConverter : ResourceEnumConverter
    {
        /// <summary>
        /// Create a new instance of the converter using translations from the given resource manager
        /// </summary>
        public SampleEnumConverter()
            : base(typeof(SampleEnum), Properties.Resources.ResourceManager)
        {
        }
    }

}
